cmd/dagger/shell.go

```diff
@@ -1472,201 +1812,176 @@ func (h *shellCallHandler) registerBuiltins() { //nolint:gocyclo
 			},
 		},
 		&ShellCommand{
-			Use:   ".doc [function]",
-			Short: "Show documentation for a type, or a function",
-			Args:  MaximumArgs(1),
+			Use: ".doc [module]\n<function> | .doc [function]",
```

Comment:
This feels like a good start of where we could start splitting these out into separate files - the implementations of some of these are now quite long, and having the very deep indentation here is hard to read.